testing/internal/e2e/host/aws/dynamichostcatalog_test.go

```diff
@@ -0,0 +1,382 @@
+	}
+
+	return nil
+}
+
+func loadConfig() (*config, error) {
+	var c config
+	err := envconfig.Process("", &c)
+	if err != nil {
+		return nil, err
+	}
+
+	return &c, err
+}
+
+func TestCreateAwsDynamicHostCatalogCli(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+
+	c, err := loadConfig()
+	require.NoError(t, err)
+	err = c.validate()
+	require.NoError(t, err)
+
+	boundary.AuthenticateCli(t)
+
+	// Create an org and project
+	newOrgId := boundary.CreateNewOrgCli(t)
+	t.Logf("Created Org Id: %s", newOrgId)
+	newProjectId := boundary.CreateNewProjectCli(t, newOrgId)
+	t.Logf("Created Project Id: %s", newProjectId)
+
+	// Create a dynamic host catalog
+	output := e2e.RunCommand("boundary", e2e.WithArgs("host-catalogs", "create", "plugin",
+		"-scope-id", newProjectId,
+		"-plugin-name", "aws",
+		"-attr", "disable_credential_rotation=true",
+		"-attr", "region=us-east-1",
+		"-secret", "access_key_id=env://E2E_AWS_ACCESS_KEY_ID",
+		"-secret", "secret_access_key=env://E2E_AWS_SECRET_ACCESS_KEY",
+		"-format", "json",
+	))
+	require.NoError(t, output.Err, string(output.Stderr))
+	var newHostCatalogResult hostcatalogs.HostCatalogCreateResult
+	err = json.Unmarshal(output.Stdout, &newHostCatalogResult)
+	require.NoError(t, err)
+	newHostCatalogId := newHostCatalogResult.Item.Id
+	t.Logf("Created Host Catalog: %s", newHostCatalogId)
+
+	// Create a host set
+	output = e2e.RunCommand("boundary", e2e.WithArgs("host-sets", "create", "plugin",
+		"-host-catalog-id", newHostCatalogId,
+		"-attr", "filters="+c.AwsHostSetFilter1,
+		"-name", "e2e Automated Test Host Set",
+		"-format", "json",
+	))
+	require.NoError(t, output.Err, string(output.Stderr))
+	var newHostSetResult hostsets.HostSetCreateResult
+	err = json.Unmarshal(output.Stdout, &newHostSetResult)
+	require.NoError(t, err)
+	newHostSetId1 := newHostSetResult.Item.Id
+	t.Logf("Created Host Set: %s", newHostSetId1)
+
+	// Get list of hosts in host set
+	// Retry is needed here since it can take a few tries before hosts start appearing
+	t.Logf("Looking for items in the host set...")
+	var actualHostSetCount1 int
+	err = backoff.RetryNotify(func() error {
+		output = e2e.RunCommand("boundary", e2e.WithArgs("host-sets", "read",
+			"-id", newHostSetId1,
+			"-format", "json",
+		))
+		if output.Err != nil {
+			return backoff.Permanent(errors.New(string(output.Stderr)))
+		}
+
+		var hostSetsReadResult hostsets.HostSetReadResult
+		err = json.Unmarshal(output.Stdout, &hostSetsReadResult)
+		if err != nil {
+			return backoff.Permanent(err)
+		}
+
+		actualHostSetCount1 = len(hostSetsReadResult.Item.HostIds)
+		if actualHostSetCount1 == 0 {
+			return errors.New("No items are appearing in the host set")
+		} else {
+			t.Logf("Found %d hosts", actualHostSetCount1)
+		}
+
+		return nil
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(3*time.Second), 5), func(err error, td time.Duration) {
+		t.Logf("%s. Retrying...", err.Error())
+	})
```

Comment:
Minor formatting nit: I prefer having all arguments of a function at the same indentation. Here's a suggested change:
```suggestion
	err = backoff.RetryNotify(
		func() error {
			output = e2e.RunCommand("boundary", e2e.WithArgs("host-sets", "read",
				"-id", newHostSetId1,
				"-format", "json",
			))
			if output.Err != nil {
				return backoff.Permanent(errors.New(string(output.Stderr)))
			}

			var hostSetsReadResult hostsets.HostSetReadResult
			err = json.Unmarshal(output.Stdout, &hostSetsReadResult)
			if err != nil {
				return backoff.Permanent(err)
			}

			actualHostSetCount1 = len(hostSetsReadResult.Item.HostIds)
			if actualHostSetCount1 == 0 {
				return errors.New("No items are appearing in the host set")
			} else {
				t.Logf("Found %d hosts", actualHostSetCount1)
			}

			return nil
		},
		backoff.WithMaxRetries(backoff.NewConstantBackOff(3*time.Second), 5),
		func(err error, td time.Duration) {
			t.Logf("%s. Retrying...", err.Error())
		},
	)
```